### PR TITLE
Fix #288348 - improved Play Panel layout (more compact)

### PR DESCRIPTION
**Pull request by Obliquely**
Suggested adjustments to allow the Play Panel to be less wide and also less tall and, arguably, look a little bit neater.

The panel's minimum width increased in 3.1beta as compared to 3.05 and and https://github.com/musescore/MuseScore/pull/4609 had the side effect of making the minimum even wider). So this change addresses those issues.

Of course, UI changes can be controversial! The approach suggested by fmiyara  in #288348 was just a graphical mockup. This PR implements that so folk can assess whether it's a good change or not.

PlayPanel

### CODE REVIEW

**Comment by IsaacWeiss** [comment]
Maybe for consistency the volume and tempo should both have the editable field in percentage points, and show "x BPM" and "y dB" in the label?

**Comment by Obliquely** [comment]
While a big fan of consistency, in this case I'm not quite so sure. 50% or 200% make a lot of sense for tempo, but less so for volume. (I had a quick scout round some other music apps to see if there were examples of that - not seen so far.) 

To be fair, I'm not all the sure that being able to enter volume (either as % or dB) is really valuable - maybe there is a solution that takes out the volume edit box. Of course, once a feature is "in" - even in beta - it can feel like a shame to take it away - and it'd be interesting if there are other views on that recent change.

**Comment by RobFog** [comment]
I think you have made some great improvements!

**Positives**
The box at the top of the panel always seemed pointless and mildly confusing to me. Good to see it gone. At the same time, the labels "Bar.Beat" and "Time" add clarity.

Also I think the layout of the controls makes more sense. Having the more important buttons, "back" and "play", on the left seems more intuitive. And I agree the BPM label should not be in that row.

**Could be better?**
The bottom line of labels seems clearer to me in the 3.1 beta panel. Perhaps we should afford two lines of labels there to accommodate the "Tempo" label and "Metronome volume" instead of just "Metronome"?

**Question**
Do the sliders become taller when the panel is made larger vertically?

File: mscore/playpanel.ui [hunk 1]
```diff
@@ -47,273 +47,251 @@
...
-              <set>Qt::NoTextInteraction</set>
-             </property>
-            </widget>
-           </item>
           </layout>
          </item>
+         <item>
+          <widget class="QLabel" name="label">
+           <property name="text">
+            <string>Bar.Beat: </string>
```
Comment (Jojo-Schmitz) [inline mscore/playpanel.ui]: `Measure.Beat`, US English please

File: mscore/playpanel.ui [hunk 2]
```diff
@@ -531,94 +575,83 @@
...
+        <property name="focusPolicy">
+         <enum>Qt::TabFocus</enum>
         </property>
-        <property name="textInteractionFlags">
-         <set>Qt::NoTextInteraction</set>
+        <property name="toolTip">
+         <string>Metronome volume</string>
+        </property>
+        <property name="accessibleName">
+         <string>Metronome Volume</string>
```
Comment (Jojo-Schmitz) [inline mscore/playpanel.ui]: better change to `Metronome volume`, same string as the tooTip, makes translation easier as it gets detected as a duplicate and the 'recycles' the translation, i.e. only one transltion for 2 strings. And capitalization isn't important at all for accessibility texts, i.e. for screen readers

File: mscore/playpanel.ui [hunk 3]
```diff
@@ -531,94 +575,83 @@
...
+        </property>
+        <property name="accessibleDescription">
+         <string>Use up and down arrows to change value</string>
```
Comment (Jojo-Schmitz) [inline mscore/playpanel.ui]: not sure we use accessibleDescription anywhere?

File: mscore/playpanel.ui [hunk 4]
```diff
@@ -479,24 +487,60 @@
...
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="toolTip">
-         <string>Metronome volume</string>
+         <string>Relative tempo</string>
         </property>
         <property name="accessibleName">
-         <string>Metronome Volume</string>
+         <string>Relative Tempo to 120 beats per minute</string>
```
Comment (Jojo-Schmitz) [inline mscore/playpanel.ui]: That is more a toolTip than an accessibilty info, so better swap those 2

File: mscore/playpanel.ui (hunk 1 above)
Comment (Obliquely) [reply mscore/playpanel.ui]: Thanks. Have revised label to read "Position". Turns out "Measure.Beat" is quite wide and makes it difficult to keep the width compact.

File: mscore/playpanel.ui (hunk 2 above)
Comment (Obliquely) [reply mscore/playpanel.ui]: Thanks. Done.

File: mscore/playpanel.ui (hunk 3 above)
Comment (Obliquely) [reply mscore/playpanel.ui]: This was in the existing setup. I've left it in for now, but it can be taken out if it's redundant.

File: mscore/playpanel.ui (hunk 4 above)
Comment (Obliquely) [reply mscore/playpanel.ui]: Thanks. Not quite sure what was going on. And the relative to "120" wouldn't be true anyway given how the Play Panel currently works. Fixed.

File: mscore/playpanel.cpp [hunk 5]
```diff
@@ -60,6 +60,7 @@ PlayPanel::PlayPanel(QWidget* parent)
       volLabel();
       volSpinBox->setRange(-80,0);
       volSpinBox->setValue(-40.0);
+      volSpinBox->setDecimals(1);
```
Comment (Jojo-Schmitz) [inline mscore/playpanel.cpp]: Might better be sone in the ui file

**Comment by Obliquely** [comment]
Thanks for feedback @RobFog and @Jojo-Schmitz. There were also some other comments on the issue on musecore.org that I've taken into account.  @RobFog on query about sliders - yes, they stretch. A tall, wide and compact screenshot:

tall (B)

wide (B)

most compact (B)

File: mscore/playpanel.cpp (hunk 5 above)
Comment (Obliquely) [reply mscore/playpanel.cpp]: OK. Makes sense. And, in fact, it was already set to 1 in UI file, so I've just removed the redundant line of code. Thanks.

**Comment by RobFog** [comment]
I'm a fan! Looks very good.

How is capitalisation used for labels? I feel "Volume" should have a lower-case "v" but I don't know what the conventions are…

**Comment by Obliquely** [comment]
Changed slider labels to sentence case. The style guidance is (arguably) ambiguous in a case like this, as the soldier's aren't quite Combo Box or Check Box labels. Image of the change attached. 

Play Panel (Sentence Case)

For comparison, the mixer uses Title Case with a vertical slider. So that ought to change. (Or they both go Title. **Either way** the Style Guide could be usefully amended to explicitly cover this case, which I am happy to once work on PR is finalised.)

Mixer (Title Case)

**Comment by anatoly-os** [comment]
@Obliquely hello and thanks for very good UX changes!

Would you mind squashing the commits to have only one here? `git rebase -i HEAD~6` should help.

Regarding the capitalisation, the Qt guide suggests using Sentence casing for labels, so this part is good (https://doc-snapshots.qt.io/qtcreator-extending/qtcreator-ui-text.html#ui-text-capitalization).

**Comment by Obliquely** [comment]
@anatoly-os  thanks and hello back. All commits squashed. I spotted a missing translation ("Tempo") too so fixed that before the squash.

File: mscore/playpanel.cpp [hunk 6]
```diff
@@ -221,7 +221,7 @@ void PlayPanel::setEndpos(int val)
 void PlayPanel::setTempo(double val)
       {
       int tempo = lrint(val * 60.0);
-      tempoLabel->setText(QString("%1 BPM").arg(tempo, 3, 10, QLatin1Char(' ')));
+      tempoLabel->setText(QString(tr("Tempo") +"\n%1 " + tr("BPM")).arg(tempo, 3, 10, QLatin1Char(' ')));
```
Comment (Jojo-Schmitz) [inline mscore/playpanel.cpp]: could well be `tr("Tempo\n%1 BPM").arg(tempo, 3, 10, QLatin1Char(' ')` too

Comment (Obliquely) [reply mscore/playpanel.cpp]: OK. Thanks. Changed and squashed.

**Comment by anatoly-os** [comment]
Thank you again. The only thing could be improved here is to keep Play raw buttons relative position and keep them center aligned so that when user resize the play panel horizontally, the buttons won't move too far from each other.

Small Play panel:
Play Panel 2019-05-21 11-09-08

Wide Play Panel:
Play Panel 2019-05-21 11-09-59

**Comment by RobFog** [comment]
Hi @Obliquely, almost two years on I just want to say your improved Play Panel is a joy to use. It's so much better than what MuseScore had before! 🎉

**Comment by RobFog** [comment]
You're welcome.

> I’m very much an active MS3 user.

Me too!

>Though varying playback speed doesn't seem to be in MS4 yet.

Oh, that'd be terrible… I reckon educators and others who use MuseScore for practice use that all the time.

**Comment by cbjeukendrup** [comment]
> > Though varying playback speed doesn't seem to be in MS4 yet.
> 
> Oh, that'd be terrible… I reckon educators and others who use MuseScore for practice use that all the time.

It does work in MS4: undock the playback toolbar to reveal the tempo percentage slider.
